compiler/src/compiler.rs

```diff
@@ -21,8 +23,8 @@ impl Compiler {
     }
   }
 
-  pub fn throw(&mut self, error: CompilerErrorKind, index: usize) {
-    self.errors.push(CompilerError::new(error, index))
+  pub fn throw(&mut self, error: CompilerErrorKind, range: Range) {
+    self.errors.push(CompilerError::new(error, range))
   }
```

Comment:
This wouldn't be needed with the solution proposed above.

Reply:
Resolved by 536acff5dde53dfc5fe80af859009e84b9ab5156